webapp/channels/src/utils/text_formatting.tsx

```diff
@@ -704,6 +715,79 @@ export function highlightCurrentMentions(
     return output;
 }
 
+export function highlightWithoutNotificationKeywords(
+    text: string,
+    tokens: Tokens,
+    highlightKeys: HighlightWithoutNotificationKey[] = [],
+) {
+    let output = text;
+
+    // Store the new tokens in a separate map since we can't add objects to a map during iteration
+    const newTokens = new Map();
+
+    // Look for highlighting keywords in the tokens
+    tokens.forEach((token, alias) => {
+        const tokenOriginalText = token.originalText.toLowerCase();
+
+        if (highlightKeys.findIndex((highlightKey) => highlightKey.key.toLowerCase() === tokenOriginalText) !== -1) {
+            const newIndex = tokens.size + newTokens.size;
+            const newAlias = `$MM_HIGHLIGHTKEYWORD${newIndex}$`;
+
+            newTokens.set(newAlias, {
+                value: `<span class="non-notification-highlight">${alias}</span>`,
+                originalText: token.originalText,
+            });
+            output = output.replace(alias, newAlias);
+        }
+    });
+
+    // Copy the new tokens to the tokens map
+    newTokens.forEach((newToken, newAlias) => {
+        tokens.set(newAlias, newToken);
+    });
+
+    // Look for highlighting keywords in the text
+    function replaceHighlightKeywordsWithToken(
+        _: string,
+        prefix: string,
+        highlightKey: string,
+        suffix = '',
+    ) {
+        const index = tokens.size;
+        const alias = `$MM_HIGHLIGHTKEYWORD${index}$`;
+
+        // Set the token map with the replacement value so that it can be replaced back later
+        tokens.set(alias, {
+            value: `<span class="non-notification-highlight">${highlightKey}</span>`,
+            originalText: highlightKey,
+        });
+
+        return prefix + alias + suffix;
+    }
+
+    highlightKeys.
+        sort((a, b) => b.key.length - a.key.length).
```

Comment:
I forget where we landed on fixing handling overlapping highlights since we discussed that last week. Did we decide this method of fixing it is enough for now?

Reply:
yes will open a ticket for the case